src/Components/Admin/EnvironmentEditor/transitionModal.js

```diff
@@ -0,0 +1,87 @@
+import React, { useEffect } from "react";
+import DialogTitle from "@material-ui/core/DialogTitle";
+import DialogContent from "@material-ui/core/DialogContent";
+import DialogActions from "@material-ui/core/DialogActions";
+import Dialog from "@material-ui/core/Dialog";
+import { Button, IconButton } from "@material-ui/core";
+import KeyboardArrowUp from "@material-ui/icons/KeyboardArrowUp";
+import KeyboardArrowDown from "@material-ui/icons/KeyboardArrowDown";
+
+export default function TransitionModal({
+    modalOpen,
+    handleModalClose,
+    handleSubmit,
+    originalTransitions,
+}) {
+    const [transitions, setTransitions] = React.useState([]);
+
+    useEffect(() => {
+        setTransitions(originalTransitions);
+    }, [originalTransitions]);
+
+    const reorder = (transitions, startIndex, endIndex) => {
+        const result = transitions;
+        const [removed] = result.splice(startIndex, 1);
+        result.splice(endIndex, 0, removed);
+        return result;
+    };
+
+    const reorderTransitions = (sourceIndex, destinationIndex) => {
+        if (
+            sourceIndex == null ||
+            destinationIndex == null ||
+            sourceIndex === destinationIndex
+        ) {
+            return;
+        }
+
+        const reorderedList = reorder(
+            transitions,
+            sourceIndex,
+            destinationIndex
+        );
+
+        setTransitions([...reorderedList]);
+    };
+
+    const onMoveUpClick = (index) => {
+        reorderTransitions(index, Math.max(0, index - 1));
+    };
+
+    const onMoveDownClick = (index) => {
+        reorderTransitions(index, Math.min(transitions.length - 1, index + 1));
+    };
+
+    return (
+        <Dialog open={modalOpen} onClose={handleModalClose}>
+            <DialogTitle>Edit Transitions Order</DialogTitle>
+            <DialogContent>
+                {transitions.map((transition, index) => {
+                    return (
+                        <div key={transition.id}>
+                            <h4>{transition.text}</h4>
+                            <h5>
+                                Transition {index + 1} of {transitions.length}
```

Comment:
Could we reverse the order

Reply:
What do you mean? Want transition 1 on bottom? I feel like the current order makes more sense

Reply:
No I mean Transition {index + 1} of {transitions.length} above the actual text

Reply:
Oh I see. will do